plugin/parser/differ/pg/differ.go

```diff
@@ -32,22 +32,25 @@ type diffNode struct {
 	dropNodeList    []ast.Node
 }
 
+type constraintMap map[string]*constraintInfo
 type tableMap map[string]*tableInfo
 type schemaMap map[string]*schemaInfo
 
 type schemaInfo struct {
-	id           int
-	existsInNew  bool
-	createSchema *ast.CreateSchemaStmt
-	tableMap     tableMap
+	id            int
+	existsInNew   bool
+	createSchema  *ast.CreateSchemaStmt
+	tableMap      tableMap
+	constraintMap constraintMap
```

Comment:
IMO, the `constraintMap` should be in `tableInfo` instead of `schemaInfo`

Reply:
I put the `constraintMap` in the `schemaInfo` instead of the `tableInfo` because the names of the constraints are unique in on schema namespace. But put it in `tableInfo` is ok for me. Are there any other concerns besides the objects relationship?

Reply:
Nope, the constraint names are unique in the table namespace. Please check with the `CHECK constraint`. The difference from the `Unique` and `Primary Key` is that they will create the index with the same name as the constraint name and index names are unique in the schema namespace.

Reply:
Thx. I will fix it in the following PR.

Reply:
Fix it in #3364